.github/workflows/terraform.yaml

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Terraform checks
+
+on:
+  workflow_call:
+    inputs:
+      module-directory:
+        description: "Directory where the terraform module is hosted in the repo"
+        default: "./terraform"
+        required: false
+        type: string
+
+jobs:
+  format:
+    name: Terraform format
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.module-directory }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Setup Terraform and Just
+        run: |
+          sudo snap install terraform --classic
+          sudo snap install just --classic
+
+      - name: Run terraform format
+        run: |
+          just validate
+          just fmt
+
+  lint:
+    name: Terraform lint
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.module-directory }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          set +e
+          sudo snap install terraform --classic
+          sudo snap install tflint
+          sudo snap install just --classic
+
+      - name: Run tflint
+        run: just lint
+
+  apply:
+    name: Terraform apply
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.module-directory }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up operator environment
+        run: |
+          set +e
+
+          sudo apt-get purge $(dpkg -l | grep -i docker | awk '{print $2}')
+          sudo apt-get purge containerd.io
+          sudo rm -r /run/containerd
+
+          sudo snap install concierge --classic
+          sudo concierge prepare -p k8s
+
+          sudo snap install docker
```

Comment:
Tiny improvement: a conditional to install docker or not, it can even be a diff step:

```
- name: Install docker if required
   if: <some-input-value-is-true>
   run: |
     sudo snap install docker
```